packages/adblocker/src/engine/engine.ts

```diff
@@ -325,20 +342,26 @@ export default class FilterEngine extends EventEmitter<EngineEventHandlers> {
           }
         }
       }
+    }
 
-      for (const [name, resource] of engine.resources.resources) {
-        if (!resources.has(name)) {
-          resources.set(name, resource);
+    if (skipResources === true) {
+      resources = new Resources();
+    } else {
+      for (const engine of engines.slice(1)) {
+        if (engine.resources.checksum !== engines[0].resources.checksum) {
+          throw new Error(
+            `resource checksum of all merged engines must match with the first one: "${engines[0].resources.checksum}" but got: "${engine.resources.checksum}"`,
+          );
         }
       }
+      const resourcesBuffer = StaticDataView.allocate(engines[0].resources.getSerializedSize(), {
```

Comment:
Nit. I wonder if there is an easier way to do a copy?

Reply:
how about this?

```ts
    engine.resources = new Resources(structuredClone(resources));
```

Reply:
Maybe `Resources` could implement its own copy (as far as I see it's a bunch of Maps so it should be very little code that does not need to be super generic)